tests/components/ring/test_light.py

```diff
@@ -119,16 +105,17 @@ async def test_light_errors_when_turned_on(
 
     assert not any(config_entry.async_get_active_flows(hass, {SOURCE_REAUTH}))
 
-    with patch.object(
-        ring_doorbell.RingStickUpCam, "lights", new_callable=PropertyMock
-    ) as mock_lights:
-        mock_lights.side_effect = exception_type
-        with pytest.raises(HomeAssistantError):
-            await hass.services.async_call(
-                "light", "turn_on", {"entity_id": "light.front_light"}, blocking=True
-            )
-        await hass.async_block_till_done()
-    assert mock_lights.call_count == 1
+    front_light_mock = mock_ring_devices.get_device(765432)
+    p = PropertyMock(side_effect=exception_type)
+    type(front_light_mock).lights = p
```

Comment:
why is this?

Reply:
This is the way to set a `PropertyMock` on a mock (strange I know), from the docs:

Because of the way mock attributes are stored you can’t directly attach a [PropertyMock](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.PropertyMock) to a mock object. Instead you can attach it to the mock type object:
```
>>>
m = MagicMock()
p = PropertyMock(return_value=3)
type(m).foo = p
m.foo
3
p.assert_called_once_with()
```

Reply:
can't you use something like
```py
front_light_mock.lights.side_effect = exception_type
```
or
```py
front_light_mock.return_value.lights.side_effect = exception_type
```
One of these might probably work

Reply:
That doesn't work because it actually runs the mock when you try to access it. This probably explains it better:
![image](https://github.com/home-assistant/core/assets/51370195/e194d7ab-ecda-4578-9e66-fff72471ee08)
N.B. `history` is a callable mock vs property.

Once the library moves to async all this PropertyMock stuff can be dropped because the rest calls will be callable rather than properties as they are now.